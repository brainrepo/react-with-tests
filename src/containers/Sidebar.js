import {Link} from 'react-router-dom'

function Sidebar() {
  return (
    <div>
      <h1>Fourviere</h1>
      <ul>
        <li>
            <Link to="/" alt="">Podcasts</Link>
        </li>
        <li>
            <Link to="/configurations" alt="">Configurations</Link>
        </li>
      </ul>
    </div>
  );
}

export default Sidebar;
