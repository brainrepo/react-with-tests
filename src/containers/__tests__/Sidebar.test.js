import { render, screen } from "@testing-library/react";
import Sidebar from "../Sidebar";
import { BrowserRouter as Router } from "react-router-dom";

describe("<Sidebar/>", () => {
  it("can be loaded", () => {
    render(
      <Router>
        <Sidebar />
      </Router>
    );
    const sidebar = screen.getByText("Fourviere");
    expect(sidebar).toBeVisible();
  });
  it("can match the snapshot", () => {
    const fragment = render(
      <Router>
        <Sidebar />
      </Router>
    ).asFragment();
    expect(fragment).toMatchSnapshot();
  });
});
