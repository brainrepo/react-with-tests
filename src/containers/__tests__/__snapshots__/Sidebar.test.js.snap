// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar/> can match the snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Fourviere
    </h1>
    <ul>
      <li>
        <a
          alt=""
          href="/"
        >
          Podcasts
        </a>
      </li>
      <li>
        <a
          alt=""
          href="/configurations"
        >
          Configurations
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
